apps/srt-tunnel.cpp

```diff
@@ -632,7 +632,7 @@ void TcpMedium::CreateListener()
 
     sockaddr_any sa = CreateAddr(m_uri.host(), m_uri.portno());
 
-    int stat = ::bind(m_socket, sa.get(), sizeof sa);
+    int stat = ::bind(m_socket, sa.get(), sizeof(*sa.get()));
```

Comment:
Maybe `sockaddr_any::size()` can be used here as well. @ethouris What do you think?
```suggestion
    int stat = ::bind(m_socket, sa.get(), sa.size());
```

Reply:
I'm surprised to see that Mac checks the size to be exactly matching - but still, I had Mac problems already as it has one more non-standard field for the size, which is still accepted with value 0. The implementation of `sockaddr_any` ignores it (as it's non-standard), but due to that I had to change the comparison to simply do it explicitly on address and port, not whole object by memory.

I think this fix is more correct because `sa.get()` returns `sockaddr*`, which isn't guaranteed to be of the right size. If `bind` is so pedantic on Mac, then it definitely needs the size that is `sizeof (sockaddr_in)` or `sizeof (sockaddr_in6)` depending on the value in `sa.get()->sa_family`. And this is what `sa.size()` returns, as long as it is properly initialized.

Reply:
@liuyanhit Please test if this works, though. I'll try to test it on my Mac as well, but I remember that Macs tend to also have platform differences between versions.

Reply:
> Maybe `sockaddr_any::size()` can be used here as well. @ethouris What do you think?

sockaddr_any::size() is better ,  when using ipv6, socket should use the correct family, while it is hard code "AF_INET" before

Reply:
I have checked ipv6 and ipv4, both work well